SillyBusinessInc/SillyBusinessGame
# DO NOT MERGE: Feat/#38 door controller

## Purpose of this PR:
Please check the implemented functionality, this is not the full ticket so DO NOT MERGE.

What this PR does is create a door prefab (RoomTransitionDoor), that will change material on being lock/unlocked, it has a little opening animation, it will disable itself if there is no SceneAsset referenced in connectsTo, and it can be locked and unlocked.

This is just a check to see if there is anything in this functionality/ code that needs to change. So it vcan be changed before the #40 and #36 tickets are finished. Those tickets are needed for further functionality of this door.

## How to test:
Open up the DoorControlScene,
Run the scene.

On the root gameObject of the RoomTransitionDoor there are contextmenuitem functions to test the locking, unlocking and opening of the door.
Test with and without scene referenced as connectsTo, to verify the disabling of the object.

### links: 
#38 

### DO NOT MERGE
